python/pyspark/sql/functions.py

```diff
@@ -2087,10 +2087,13 @@ def _wrapped(self):
         """
         Wrap this udf with a function and attach docstring from func
         """
-        @functools.wraps(self.func)
+        assignments = tuple(a for a in functools.WRAPPER_ASSIGNMENTS if a != "__name__")
```

Comment:
I'd put a comment here saying what this is for just so that when we see this next year we don't forget why we stripped out name from the things we asked functools to assign.